wallet/src/recovering/daedalus.rs

```diff
@@ -42,17 +42,21 @@ impl RecoveringDaedalus {
     /// convenient function to parse a block and check for owned token
     pub fn check_blocks<'a>(&mut self, fragments: impl Iterator<Item = &'a Fragment>) {
         for fragment in fragments {
-            let fragment_id = fragment.hash();
-            if let Fragment::OldUtxoDeclaration(utxos) = fragment {
-                for (output_index, (address, value)) in utxos.addrs.iter().enumerate() {
-                    let pointer = UtxoPointer {
-                        transaction_id: fragment_id,
-                        output_index: output_index as u8,
-                        value: *value,
-                    };
+            self.check_single_block(fragment)
+        }
+    }
+
+    pub fn check_single_block(&mut self, fragment: &Fragment) {
```

Comment:
```suggestion
    pub fn check_fragment(&mut self, fragment: &Fragment) {
```

and I would rename `check_blocks` to `check_fragments`.

Reply:
yeah, I thought about it too and chose the bad option.